Dismemberment implementation

Badulundabad opened the pull request.

Z3rca reviewed (approved):
I was at first confused about the weapons.xml change but, I think it just enhance the readability. The rest seems to work like I orginally intended, with customizability of XML files. thats great.

In General, I would suggest, if possible to make smaller commits, but that is just a personal preference.

hunharibo commented:
Yeah sorry, its not a really clean commit :( I left some experimentation in it. Will try to pay more attention in the future.

tristanka commented on HarmonyPatches/CharacterCreationPatches.cs:
```diff
@@ -0,0 +1,51 @@
+            CharacterCreationState gameState = Game.Current.GameStateManager.CreateState<CharacterCreationState>(new object[]
+{
+                new TOWCharacterCreationContent()
+});
```
Indentation is a little messed up here

tristanka commented on CharacterCreation/TOWCharacterCreationContent.cs:
```diff
@@ -0,0 +1,149 @@
+                            break;
+                    }
+                    
+                    var relevantOptions = _options.FindAll(x => x.StageNumber == i && x.Culture == culture);
```
`x.Culture == culture` -> `x.Culture.Equals(culture)` just to ensure value equality and avoid any weird reference inequalities in the future? Probably wouldn't happen but we really just care about value equality

tristanka reviewed (commented):
Also, didn't really look into why this is happening, but the culture list only shows the two custom cultures for New Campaign as well, but when you move forward, it uses the base TaleWorlds character creation content. I don't think this is a big issue right now, just wanted to point it out in case you hadn't noticed.

tristanka reviewed (approved):
🎉

Badulundabad committed:
Initial commit

Added decapitation;
There is a problem when setting the invisibility of the chopped off head it affects the heads of the rest of the agents who use this mesh.

Badulundabad committed:
Fix the GetHeadCopy method

Now this method returns the head with helmet or hat;
Fixed a bug when head meshes of all agents disappeared when getting a head mesh of a victim.
